### ISSUE TITLE
Clarify the behavior of setting ItemSize on Virtualize

### ISSUE THREAD

**Issue by danroth27** [dotnet/AspNetCore.Docs#21314]
The docs don't say much about setting the ItemSize on Virtualize. I believe Virtualize will try to estimate the ItemSize, but we don't explain how that works. What if you don't set it? What if you set it, but the size is slightly off?

@pranavkm @SteveSandersonMS 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0f4ab8b4-4d2e-9e1b-793e-dbb17f6d694d
* Version Independent ID: 26ed2425-63ec-e206-edcb-e51f17edb844
* Content: [ASP.NET Core Blazor component virtualization](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/virtualization?view=aspnetcore-5.0#item-provider-delegate)
* Content Source: [aspnetcore/blazor/components/virtualization.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/components/virtualization.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by SteveSandersonMS** [comment]
Regardless of whether you set `ItemSize`, the `<Virtualize>` component will measure the actual rendered size and use that instead of any value you give. However, this only applies *after* it's rendered the first batch of them, because until that point there aren't any to measure.

It's advantageous to tell it the true item size in advance, because then it will load the correct set for the first batch, instead of potentially loading the wrong set, seeing that they render off-screen, and triggering a second load and re-render. Additionally, if you want the browser's scroll position to be maintained accurately if you reload the page while in the middle of a virtualized list, you will need that first render to be correct otherwise the browser's scroll-position preservation behavior will put the user in the wrong place.

So in summary, setting a correct `ItemSize` helps both with avoiding unnecessary initial renders, and with correct preservation of scroll position during page reloads. If you get the size wrong, your app may behave less efficiently and may not preserve scroll position accurately during reload.

**Comment by daveabrock** [comment]
Hey, @guardrex - I'm working with `<Virtualize>` a lot now, want me to do this update? I could do it sometime next week.

**Comment by guardrex** [comment]
Sure thing, thanks @daveabrock.

There's already an Item Size section for the content ...

https://docs.microsoft.com/aspnet/core/blazor/components/virtualization#item-size

**Comment by daveabrock** [comment]
Sounds good. I need to get my `ItemSize` working for the talk I give in two weeks so I will be incentivized :)
